src/zarr/core/group.py

```diff
@@ -188,6 +284,37 @@ async def open(
             zgroup = json.loads(zgroup_bytes.to_bytes())
             zattrs = json.loads(zattrs_bytes.to_bytes()) if zattrs_bytes is not None else {}
             group_metadata = {**zgroup, "attributes": zattrs}
+
+            if open_consolidated:
```

Comment:
This could probably use a refactor. I think starting at the `if zarr_format == 2` we're all done with I/O, so we could have another constructor like `def _from_raw(...)` that doesn't have to be async.